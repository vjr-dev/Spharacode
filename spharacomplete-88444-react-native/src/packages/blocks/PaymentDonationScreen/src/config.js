// Customizable Area Start
Object.defineProperty(exports, "__esModule", {
    value: true,
});
exports.paymentDonationEndPoint = "bx_block_donationpayments/donationpayments";
// Customizable Area End