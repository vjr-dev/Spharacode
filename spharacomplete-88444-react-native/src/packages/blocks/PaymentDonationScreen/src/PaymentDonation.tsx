// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import React from "react";
import {
  Dimensions,
  Image,
  ImageBackground,
  KeyboardAvoidingView,
  Modal,
  Platform,
  SafeAreaView,
  StatusBar,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import { RFValue } from "react-native-responsive-fontsize";
import { heightPercentageToDP as hp } from "react-native-responsive-screen";
import { COLORS } from "../../../framework/src/Globals";
import {
  apple_pay,
  back1,
  back2,
  google_pay,
  image_back,
  next,
  tick,
} from "./assets";
import PaymentDonationController, { Props } from "./PaymentDonationController";
const windowWidth = Dimensions.get("window").width;
const windowHeight = Dimensions.get("window").height;

export default class PaymentDonation extends PaymentDonationController {
  constructor(props: Props) {
    super(props);
  }

  render() {
    return (
      <SafeAreaView style={styles.container1}>
        <View style={styles.child}>
          <ImageBackground source={back1} style={styles.image}>
            <ImageBackground source={back2} style={styles.image2}>
              <KeyboardAvoidingView
                keyboardVerticalOffset={1}
                behavior={"height"}
              >
                <StatusBar backgroundColor={"#f17234"} barStyle="default" />
                <View
                  style={{
                    height: hp("7%"),
                    width: windowWidth,
                    flexDirection: "row",
                    backgroundColor: COLORS.darkorange,
                    alignItems: "center",
                  }}
                >
                  <TouchableOpacity
                    style={{
                      width: "15%",
                      height: "90%",
                      justifyContent: "center",
                      alignItems: "center",
                    }}
                    testID="backButton"
                    onPress={() => this.goback()}
                  >
                    <Image
                      source={image_back}
                      style={{
                        height: 15,
                        width: 15,
                        tintColor: COLORS.backgroundGray,
                      }}
                    />
                  </TouchableOpacity>
                  <View
                    style={{
                      height: "100%",
                      width: "70%",
                      alignItems: "center",
                      justifyContent: "center",
                    }}
                  >
                    <Text
                      style={{
                        color: COLORS.backgroundGray,
                        fontWeight: "700",
                        fontSize: RFValue(16),
                      }}
                    >
                      Make payement of
                    </Text>
                  </View>
                  <View style={{ height: "100%", width: "15%" }} />
                </View>
                <View
                  style={{
                    height: 120,
                    width: windowWidth,
                    backgroundColor: COLORS.darkorange,
                    alignItems: "center",
                    justifyContent: "center",
                    alignSelf: "center",
                  }}
                >
                  <Text style={{ color: COLORS.white, fontSize: 35 }}>
                    â‚¹ {this.state.Amount}
                  </Text>
                  {!this.state.paymentdone ? (
                    <View
                      style={{
                        backgroundColor: COLORS.orangelight,
                        paddingHorizontal: 20,
                        borderRadius: 50,
                        alignItems: "center",
                        justifyContent: "center",
                        marginTop: 5,
                      }}
                    >
                      <Text style={{ fontSize: 15 }}>pending..</Text>
                    </View>
                  ) : null}
                </View>

                <View
                  style={{
                    height: 50,
                    width: windowWidth - 20,
                    alignSelf: "center",
                    alignItems: "center",
                    justifyContent: "center",
                    marginTop: 25,
                  }}
                >
                  <Text style={{ color: COLORS.ultralightwhite }}>
                    Send payment using any of the following methods.
                  </Text>
                </View>
                <TouchableOpacity
                  testID="gPayButton"
                  onPress={() => this.gpayclick()}
                  style={{
                    height: 60,
                    width: windowWidth - 20,
                    alignSelf: "center",
                    backgroundColor: COLORS.Viewback,
                    borderRadius: 50,
                    justifyContent: "space-between",
                    flexDirection: "row",
                    marginTop: 25,
                  }}
                >
                  <Image
                    source={google_pay}
                    style={{
                      backgroundColor: COLORS.white,
                      height: 40,
                      width: 40,
                      marginLeft: 10,
                      borderRadius: 50,
                      alignSelf: "center",
                    }}
                  />

                  <Text
                    style={{
                      alignSelf: "center",
                      color: COLORS.ultralightwhite,
                      width: "65%",
                    }}
                  >
                    Google Pay
                  </Text>
                  <Image
                    source={next}
                    style={{
                      height: 30,
                      width: 30,
                      tintColor: COLORS.ultralightwhite,
                      alignSelf: "center",
                      marginRight: 10,
                      opacity: 0.5,
                    }}
                  />
                </TouchableOpacity>
                <TouchableOpacity
                  testID="rPaybutton"
                  onPress={() => this.Rpay()}
                  style={{
                    height: 60,
                    width: windowWidth - 20,
                    alignSelf: "center",
                    backgroundColor: COLORS.Viewback,
                    borderRadius: 50,
                    justifyContent: "space-between",
                    flexDirection: "row",
                    marginTop: 25,
                  }}
                >
                  <Image
                    // source={google_pay}
                    style={{
                      backgroundColor: COLORS.white,
                      height: 40,
                      width: 40,
                      marginLeft: 10,
                      borderRadius: 50,
                      alignSelf: "center",
                    }}
                  />

                  <Text
                    style={{
                      alignSelf: "center",
                      color: COLORS.ultralightwhite,
                      width: "65%",
                    }}
                  >
                    REZORPAY
                  </Text>
                  <Image
                    source={next}
                    style={{
                      height: 30,
                      width: 30,
                      tintColor: COLORS.ultralightwhite,
                      alignSelf: "center",
                      marginRight: 10,
                      opacity: 0.5,
                    }}
                  />
                </TouchableOpacity>
                <TouchableOpacity
                  testID="stripePayButton"
                  onPress={() => this._checkSessionAPIcall()}
                  style={{
                    height: 60,
                    width: windowWidth - 20,
                    alignSelf: "center",
                    backgroundColor: COLORS.Viewback,
                    borderRadius: 50,
                    justifyContent: "space-between",
                    flexDirection: "row",
                    marginTop: 25,
                  }}
                >
                  <Image
                    // source={google_pay}
                    style={{
                      backgroundColor: COLORS.white,
                      height: 40,
                      width: 40,
                      marginLeft: 10,
                      borderRadius: 50,
                      alignSelf: "center",
                    }}
                  />

                  <Text
                    style={{
                      alignSelf: "center",
                      color: COLORS.ultralightwhite,
                      width: "65%",
                    }}
                  >
                    STRIPE PAY
                  </Text>
                  <Image
                    source={next}
                    style={{
                      height: 30,
                      width: 30,
                      tintColor: COLORS.ultralightwhite,
                      alignSelf: "center",
                      marginRight: 10,
                      opacity: 0.5,
                    }}
                  />
                </TouchableOpacity>
                {Platform.OS === "ios" ? (
                  <TouchableOpacity
                    testID="applePayButton"
                    onPress={() => this.applepayclick()}
                    style={{
                      height: 60,
                      width: windowWidth - 20,
                      alignSelf: "center",
                      backgroundColor: COLORS.Viewback,
                      borderRadius: 50,
                      justifyContent: "space-between",
                      flexDirection: "row",
                      marginTop: 25,
                    }}
                  >
                    <Image
                      source={apple_pay}
                      style={{
                        backgroundColor: COLORS.white,
                        height: 40,
                        width: 40,
                        marginLeft: 10,
                        borderRadius: 50,
                        alignSelf: "center",
                      }}
                    />

                    <Text
                      style={{
                        alignSelf: "center",
                        color: COLORS.ultralightwhite,
                        width: "65%",
                      }}
                    >
                      Apple pay
                    </Text>
                    <Image
                      source={next}
                      style={{
                        height: 30,
                        width: 30,
                        tintColor: COLORS.ultralightwhite,
                        alignSelf: "center",
                        marginRight: 10,
                        opacity: 0.5,
                      }}
                    />
                  </TouchableOpacity>
                ) : null}
                <Modal transparent={true} visible={this.state.Modal}>
                  <View
                    style={{
                      height: "100%",
                      width: "100%",
                      backgroundColor: COLORS.darkorange,
                      alignItems: "center",
                      justifyContent: "center",
                      alignSelf: "center",
                    }}
                  >
                    <Image
                      style={{ height: 100, width: 100, alignSelf: "center" }}
                      source={tick}
                    />
                    <Text style={{ fontSize: 35, marginTop: 75 }}>
                      Successful!
                    </Text>
                    <Text
                      style={{
                        fontSize: 18,
                        color: COLORS.white,
                        marginTop: 5,
                      }}
                    >
                      You donate â‚¹ {this.state.Amount} to {this.state.to}
                    </Text>
                    <TouchableOpacity
                      testID="doneButton"
                      onPress={() => this.successfulclick()}
                      style={{
                        height: 50,
                        width: windowWidth - 20,
                        alignItems: "center",
                        justifyContent: "center",
                        borderWidth: 1,
                        borderRadius: 50,
                        marginTop: 75,
                        borderStyle: "dashed",
                      }}
                    >
                      <Text>DONE</Text>
                    </TouchableOpacity>
                  </View>
                </Modal>
              </KeyboardAvoidingView>
            </ImageBackground>
          </ImageBackground>
        </View>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: windowHeight,
    width: windowWidth,
    backgroundColor: COLORS.white,
  },
  container1: { flex: 1, backgroundColor: COLORS.darkorange },
  child: { width: windowWidth },

  image: {
    height: windowHeight,
    width: windowWidth,
  },
  image2: {
    height: "100%",
    width: windowWidth,
    alignItems: "center",
  },
  btnview: {
    alignItems: "center",
    justifyContent: "center",
  },
  btn: {
    backgroundColor: "#FF0100",
    height: 60,
    width: 330,
    marginTop: 80,
    bottom: 30,
    alignItems: "center",
    justifyContent: "center",
    borderRadius: 30,
  },
  btntxt: {
    fontSize: 15,
    color: "#ffffff",
    letterSpacing: 0.5,
  },
});
// Customizable Area End
