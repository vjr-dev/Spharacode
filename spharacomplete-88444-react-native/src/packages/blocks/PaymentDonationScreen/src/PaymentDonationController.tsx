// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import { BlockComponent } from "../../../framework/src/BlockComponent";
import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import MessageEnum, {
  getName,
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";

import AsyncStorage from "@react-native-async-storage/async-storage";
import { Alert, Platform } from "react-native";
import { GooglePay } from "react-native-google-pay";
const PaymentRequest = require("react-native-payments").PaymentRequest;

import RazorpayCheckout from "react-native-razorpay";

const allowedCardNetworks: any = ["VISA", "MASTERCARD"];
const allowedCardAuthMethods: any = ["PAN_ONLY", "CRYPTOGRAM_3DS"];

const requestData: any = {
  cardPaymentMethod: {
    tokenizationSpecification: {
      type: "PAYMENT_GATEWAY",
      gateway: "stripe",
      gatewayMerchantId: "",
      stripe: {
        publishableKey: "pk_test_TYooMQauvdEDq54NiTphI7jx",
        version: "2018-11-08",
      },
    },
    allowedCardNetworks,
    allowedCardAuthMethods,
  },
  transaction: {
    totalPrice: "10",
    totalPriceStatus: "FINAL",
    currencyCode: "IND",
  },
  merchantName: "Example Merchant",
};

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
}

interface S {
  data: any;
  Amount: any;
  Modal: boolean;
  UserName: string;
  Nickname: string;
  UserNumber: any;
  UserEmail: any;
  to: string;
  STRIPE_PUBLIC_KEY: string;
  CHECKOUT_SESSION_ID: string;
  currentPosition: string;
  paymentdone: any;
}

interface SS {
  id: any;
}

export default class PaymentDonationController extends BlockComponent<
  Props,
  S,
  SS
> {
  checkSessionAPIcallId: any;

  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);
    this.subScribedMessages = [
      getName(MessageEnum.RestAPIResponceMessage),
      getName(MessageEnum.RestAPIRequestMessage),
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      getName(MessageEnum.RestAPIRequestBodyMessage),
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      getName(MessageEnum.RestAPIRequestMethodMessage),
    ];

    this.state = {
      data: [1, 2, 3, 4, 5],
      Amount: 1,
      Modal: false,
      to: "Police officers",
      STRIPE_PUBLIC_KEY: "",
      CHECKOUT_SESSION_ID: "",
      UserName: "",
      Nickname: "",
      UserNumber: "",
      UserEmail: "",
      currentPosition: "",
      paymentdone: false,
    };
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }

  async receive(from: string, message: Message) {
    runEngine.debugLog("Message Recived", message);

    if (message.id === getName(MessageEnum.RestAPIResponceMessage)) {
      const apiRequestCallId = message.getData(
        getName(MessageEnum.RestAPIResponceDataMessage)
      );

      let responseJson = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );

      if (apiRequestCallId === this.checkSessionAPIcallId) {
        if (responseJson !== null && responseJson) {
          if (responseJson.donation_payment != null) {
            this.props.navigation.navigate("StripeWebView", {
              url: responseJson.checkout_session,
            });
          }
        }
      }
    }
  }
  async componentDidMount() {
    let USERData: any = await AsyncStorage.getItem("User_Data");

    console.log("USERData",USERData)
    let temp: any = {};
    
    if(USERData!== null) //istanbul ignore next
    { 
      temp = JSON.parse(USERData);
      
    }
   
    this.setState({
      UserName: temp.data.attributes.first_name,
      Nickname: temp.data.attributes.nick_name,
      UserNumber: temp.data.attributes.phone_number,
      UserEmail: temp.data.attributes.email,
      currentPosition: temp.data.attributes.current_position,
    });
    this.setState({
      Amount: this.props.route.params?.Amount,
      to: this.props.route.params?.Dto,
    });
  }

  successfulclick() {
    this.setState({ Modal: false, paymentdone: true });
  }

  gpayclick() {
    console.log("OS--> ", Platform.OS)
    if (Platform.OS === "android") {
      GooglePay.setEnvironment(GooglePay.ENVIRONMENT_TEST);

      GooglePay.isReadyToPay(allowedCardNetworks, allowedCardAuthMethods).then(
        (ready:boolean) => {
          if (ready) //istanbul ignore next
          { 
            GooglePay.requestPayment(requestData)
              .then((token: string) => {
                
                this.setState({ Modal: true });
              })
              .catch((error: any) => console.log(error.code, error.message));
          }
        }
      );
    } else {
      Alert.alert("","Under devlopment");
    }
  }

  Rpay() {
    let options = {
      description: "Credits towards consultation",
      image: "https://i.imgur.com/3g7nmJC.png",
      currency: "INR",
      key: "rzp_test_9i47TjiFtle6se", // Your api key
      amount: this.state.Amount * 100,
      name: this.state.UserName,
      prefill: {
        email: this.state.UserEmail,
        contact: this.state.UserNumber,
        name: "Razorpay Software",
      },
      theme: { color: "#F37254" },
    };
    RazorpayCheckout.open(options)
      .then((data: any) => {
        this.setState({ Modal: true });
      })
      .catch((error: any) => //istanbul ignore next
      { 
        Alert.alert("","Payment failed please try again");
      });
  }

  applepayclick() {
    if (Platform.OS === "ios") { 
      let paymentRequest = new PaymentRequest(
        [
          {
            supportedMethods: ["apple-pay"],
            data: {
              merchantIdentifier: "merchant.com.sphara",
              supportedNetworks: ["visa", "mastercard", "amex"],
              countryCode: "US",
              currencyCode: "USD",
            },
          },
        ],
        {
          id: "basic-example",
          displayItems: [
            {
              label: this.state.to,
              amount: {
                currency: "IND",
                value: this.state.Amount,
              },
            },
          ],
          total: {
            label: "Donation",
            amount: { currency: "IND", value: this.state.Amount },
          },
        },
        {
          requestPayerName: true,
          requestPayerPhone: true,
          requestPayerEmail: true,
          requestShipping: false,
        }
      );
      paymentRequest
        .canMakePayments()
        .then((canMakePayment) => //istanbul ignore next 
        { 
          if (canMakePayment) //istanbul ignore next 
          {  
            paymentRequest
              .show()
              .then((paymentResponse: { complete: (arg0: string) => void }) => {
                paymentResponse.complete("success");
              })
              .catch((error: any) => console.log(error));
          } else //istanbul ignore next
          {
            console.log("Cant Make Payment");
          }
        })
        .catch((error: any) => //istanbul ignore next
        {console.log(error, "ERROR1")});
    }
  }
  goback() {
    this.props.navigation.pop();
  }
  _checkSessionAPIcall = async () => {
    let Token0: any = await AsyncStorage.getItem("Token");

    const header = {
      "Content-Type": "application/json",
      token: Token0,
    };
    let httpBody = JSON.stringify({
      data: {
        attributes: {
          donatelist_id: 1,
          amount: this.state.Amount,
          success_url: "https://example.com/success",
          cancel_url: "https://example.com/cancel",
        },
      },
    });
    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );
    this.checkSessionAPIcallId = requestMessage.messageId;
    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      configJSON.paymentDonationEndPoint
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      "POST"
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestBodyMessage),
      httpBody
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      header
    );
    runEngine.sendMessage(requestMessage.id, requestMessage);
    return true;
  };
}
// Customizable Area End
