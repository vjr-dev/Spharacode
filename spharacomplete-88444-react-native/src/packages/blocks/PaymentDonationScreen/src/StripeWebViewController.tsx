// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import { BlockComponent } from "../../../framework/src/BlockComponent";
import { runEngine } from "../../../framework/src/RunEngine";
import MessageEnum, {
  getName,
} from "../../../framework/src/Messages/MessageEnum";

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
}

export interface S {
}

export interface SS {
  id: any;
}

export default class StripeWebViewController extends BlockComponent<
  Props,
  S,
  SS
> {

  constructor(props: Props) {
    super(props);
    this.subScribedMessages = [
      getName(MessageEnum.RestAPIResponceMessage),
    ];
    this.receive = this.receive.bind(this);
    runEngine.attachBuildingBlock(this, this.subScribedMessages);

    this.state = {

    };
  }
}
// Customizable Area End