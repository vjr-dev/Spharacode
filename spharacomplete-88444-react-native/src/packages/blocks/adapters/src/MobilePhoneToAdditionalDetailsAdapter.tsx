import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class MobilePhoneToAdditionalDetailsAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationMobilePhoneAdditionalDetailsMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'AdditionalDetailForm'
    );

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );
    raiseMessage.addData(
      getName(MessageEnum.AuthTokenDataMessage),
      from.getData(getName(MessageEnum.AuthTokenDataMessage))
    );

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
