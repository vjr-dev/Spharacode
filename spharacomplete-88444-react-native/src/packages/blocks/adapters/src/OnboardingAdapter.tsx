import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class OnboardingAdapter {
  send: (message: Message) => void;

  constructor() {
    // const uuidv4 = require('uuid/v4');
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.OnboardingGuideControllerDoneOnboarding),
      getName(MessageEnum.OnboardingGuideControllerSkipOnboarding)
    ]);
  }

  convert = (from: Message): Message => {
    runEngine.debugLog("OnboardingAdapter", from.id)
    const to = new Message(getName(MessageEnum.NavigationMessage));
    to.addData(getName(MessageEnum.NavigationTargetMessage), 'Home');
    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );
    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
