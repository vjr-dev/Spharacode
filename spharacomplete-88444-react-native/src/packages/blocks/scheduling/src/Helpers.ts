/* App/Lib/GeneralHelpers.js */
import { Platform, StyleSheet } from 'react-native';
import { heightFromPercentage } from '../../../framework/src/Utilities';

// Colors Constants
export const COLORS = {
  TRANSPARENT: 'transparent',
  white: 'rgb(255,255,255)',
  black: 'rgb(0,0,0)',
  black10: 'rgb(10,10,10)',
  green: 'rgb(76,217,100)',
  red: 'red',
  yellow: 'rgba(255, 212, 40, 1)',
  darkGray: 'rgba(105,105,105,0.9)',
  lightGray: '#a7b6b9',
  skipGray: 'rgb(91,91,92)',
  borderColor: 'rgb(240, 240, 240)',
  inactiveIndicator: 'rgb(228, 233, 242)',
  inputLabel: 'rgb(143,155,179)',
  inputIcon: 'rgb(197,206,224)',
  inputValue: 'rgb(46,58,89)',
  discountRed: 'rgb(199,5,28)',
  orange: 'rgb(255,157,43)',
  infoGray: 'rgb(163,164,165)',
  ReviewBlack: 'rgb(34,43,69)',
  blue: 'blue',
  blueGrey: 'rgb(143,155,179)',
  darkBlueGrey: 'rgb(46,58,89)',
  lightBlueGrey: 'rgb(197, 206, 224)'
};

export const CommonStyle = StyleSheet.create({
  container: {
    flex: 1,
    width: "100%",
    justifyContent: "flex-start",
    alignItems: "center"
  },
  wrapper: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  headerContainer:{
    backgroundColor:COLORS.yellow,
    height: '10%',
  },
  authHeader:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.yellow
  },
  authHeaderWhite:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.white
  },
  formContainer: {
    width: "80%"
  },
  textAlignCenter:{
    textAlign: 'center'
  },
  labelTextStyle: {
    color: COLORS.inputLabel
  },
  inputContainerStyle:{
    borderBottomWidth: 1,
    borderBottomColor: COLORS.inactiveIndicator
  },
  error: {
    color: COLORS.red,
    fontSize: 14
  },
  scrollStyle: {
    width: "100%"
  },
  btnTitleStyle:{
    fontSize: 17,
    color: COLORS.black
  },
  closeButtonStyle: {
    backgroundColor: COLORS.inputIcon,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  yellowButtonStyle: {
    backgroundColor: COLORS.yellow,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  closeBtnTitleStyle:{
    color: COLORS.black
  },
  scrollContainer: {
    flexGrow: 1,
    alignItems: "center"
  },
  shadowStyle: {
    shadowOffset:
      Platform.OS === "ios"
        ? { width: 0, height: 1 }
        : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 3 : 5,
    elevation: 5
  },
  boxShadowStyle: {
    backgroundColor: COLORS.white,
    marginHorizontal:2,
    marginVertical:1,
    shadowOffset:
    Platform.OS === "ios"
      ? { width: 0, height: 1 }
      : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 1 : 5,
    elevation: 2,    
    borderRadius:8,
  },
  toastModal: {
    borderRadius: 20,
    marginHorizontal: 15,
    paddingHorizontal: '12%',
    paddingVertical: 100,
    alignContent: "center",
    justifyContent: "center",
  },
  clearModal: {
    borderRadius: 5,
    marginHorizontal: 50,
    height: 220,
    backgroundColor: COLORS.red
  },
  imageThumbContain: {
    resizeMode: "contain",
    width: "100%",
    height: "100%"
  },
  imageThumbCover: {
    resizeMode: "cover",
    width: "100%",
    height: "100%"
  },
  flexContainer: {
    flex: 1,
    width: "100%"
  },
  rowStyle: {
    flexDirection: "row"
  },
  flexStart: {
    justifyContent: "flex-start"
  },
  flexEnd: {
    justifyContent: "flex-end"
  },
  jCCenter: {
    justifyContent: "center"
  },
  spaceBetween: {
    justifyContent: "space-between",    
  },
  spaceAround: {
    justifyContent: 'space-around'
  },
  alignItemsCenter:{
    alignItems: "center"
  },
  styleCenter: {
    justifyContent: "flex-end",
    alignItems: "center"
  },
  centerStyle: {
    justifyContent: "center",
    alignItems: "center"
  },
  width100P:{
    width: '100%'
  },
  subView: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  emptyList: {
    alignItems: "center",
    marginTop: 100
  },
  emptyListFilter: {
    alignItems: "center",
    justifyContent: "center",
    marginTop: 100
  },
  backIconView: {
    marginVertical: 10,
    marginHorizontal: 5,
    width: 30,
    height: 30,
    justifyContent: "flex-end",
    alignItems: "flex-end"
  },
  crossImage: {
    tintColor: COLORS.red,
    width: "100%",
    height: "100%",
    resizeMode: "contain"
  },
  searchTextInput: {
    width: "80%"
  },
  errorReport: {
    width: "70%",
    color: COLORS.red,
    fontSize: 15,
    marginLeft: "28.8%"
  }
});