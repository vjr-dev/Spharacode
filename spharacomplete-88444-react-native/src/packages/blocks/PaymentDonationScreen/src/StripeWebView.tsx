// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import React from "react";
import { StyleSheet, SafeAreaView, StatusBar } from "react-native";
import StripeWebViewController, { Props } from "./StripeWebViewController";
import { WebView } from "react-native-webview";

export default class StripeWebView extends StripeWebViewController {
  constructor(props: Props) {
    super(props);
  }

  render() {
    const url = this.props.route.params?.url?.url;
    return (
      <SafeAreaView style={styles.container}>
        <StatusBar
          barStyle="light-content"
          hidden={false}
          backgroundColor="#54595F"
          translucent={true}
        />
        <WebView
          source={{
            uri: url,
          }}
        />
      </SafeAreaView>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
// Customizable Area End
