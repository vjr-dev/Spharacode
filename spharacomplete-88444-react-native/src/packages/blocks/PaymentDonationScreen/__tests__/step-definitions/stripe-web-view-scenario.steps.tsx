//@ts-ignore
//@ts-nocheck
import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";

import * as helpers from "../../../../framework/src/Helpers";
import React from "react";
import StripeView from "../../src/StripeWebView";

const screenProps = {
  navigation: {
    dispatch: jest.fn(),
    navigate: jest.fn(),
    goBack: jest.fn(),
    pop: jest.fn()
  },
  route:{
    params:{
      url:{
        url:"https://www.google.com/"
      }
    }
  },
  id: "StripeWebView"
};

const feature = loadFeature(
  "./__tests__/features/stripe-web-view-screen-scenario.feature"
);

defineFeature(feature, test => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });

  test("User navigates to Stripe Web View", ({ given, when, then }) => {
    let stripeWrapper: ShallowWrapper;
    let instance :StripeView;
    
    given("I am a User attempting to navigate Stripe Web View", () => {
      stripeWrapper = shallow(<StripeView {...screenProps} />);
      expect(stripeWrapper).toBeTruthy();
      expect(stripeWrapper).toMatchSnapshot();
    });

   

  });
});
