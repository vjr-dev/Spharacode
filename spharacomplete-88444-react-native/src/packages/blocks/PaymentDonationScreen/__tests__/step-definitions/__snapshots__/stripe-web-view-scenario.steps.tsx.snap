// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stripe Web View User navigates to Stripe Web View 1`] = `
<SafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <StatusBar
    animated={false}
    backgroundColor="#54595F"
    barStyle="light-content"
    hidden={false}
    showHideTransition="fade"
    translucent={true}
  />
  <WebView
    cacheEnabled={true}
    javaScriptEnabled={true}
    originWhitelist={
      Array [
        "http://*",
        "https://*",
      ]
    }
    source={
      Object {
        "uri": "https://www.google.com/",
      }
    }
    useSharedProcessPool={true}
  />
</SafeAreaView>
`;
